File: Android/feature-dreams-api/src/main/java/ru/iandreyshev/featureDreamsApi/domain/DreamProperties.kt
package ru.iandreyshev.featureDreamsApi.domain

import android.os.Bundle

data class DreamProperties(
        val description: String,
        val sleepingDate: SleepingDate = SleepingDate(0),
        val isLucid: Boolean = false
) {

    fun toBundle(): Bundle {
        return Bundle().apply {
            putString(KEY_DESCRIPTION, description)
        }
    }

    companion object {
        private const val KEY_DESCRIPTION = "key_description"
        private const val KEY_SLEEPING_DATE = "key_sleeping_date"

        fun create(bundle: Bundle?): DreamProperties? {
            bundle ?: return null

            val description = bundle.getString(KEY_DESCRIPTION) ?: return null

            return DreamProperties(description)
        }
    }

}

File: Backend/Server/Domain/DreamProperties.cs
using Newtonsoft.Json;

namespace Dreamland.Domain
{
	public class DreamProperties
	{
		[JsonProperty("id", NullValueHandling = NullValueHandling.Ignore)]
		public long Id { get; set; }
		[JsonProperty("description", NullValueHandling = NullValueHandling.Ignore)]
		public string Description { get; set; }
		[JsonProperty("is_lucid", NullValueHandling = NullValueHandling.Ignore)]
		public bool IsLucid { get; set; }
		[JsonProperty("sleeping_date", NullValueHandling = NullValueHandling.Ignore)]
		public long SleepingDate { get; set; }
	}
}
